fastembed/text/text_embedding_base.py

```diff
@@ -51,5 +51,7 @@ def query_embed(self, query: str, **kwargs) -> np.ndarray:
         """
 
         # This is model-specific, so that different models can have specialized implementations
-        query_embedding = list(self.embed([query], **kwargs))[0]
-        return query_embedding
+        if isinstance(query, str):
+            yield from self.embed([query], **kwargs)
+        if isinstance(query, Iterable):
```

Comment:
If we accept that, query should be union: `query: Uinion[str, Iterable[str]]`

Reply:
Done.